### ISSUE TITLE
locale must be at most 2 characters

### ISSUE THREAD

**Issue by amityweb** [DomDew/strapi-plugin-fuzzy-search#82]
We are using the 4 character locales in Strapi, e.g. fr-FR

But when using fuzzy search it wont allow it:
`/api/fuzzy-search/search/?query=c&locale=fr-FR`

`{"data":null,"error":{"status":400,"name":"BadRequestError","message":"Invalid query","details":"locale must be at most 2 characters"}}`

**Comment by amityweb** [comment]
Does not seem to work anyway, am I doing something wroing? 

The following URL returns no results even though I have a page in Spanish (es):
`http://localhost:1337/api/fuzzy-search/search?query=c&locale=es`

The following graphql returns the en/default pages:
```
query {
  search(query: "pro", locale: "es") {
    pages {
      data {
        attributes
        {
          slug
          title
        }
      }
    }
    products {
      data {
        attributes {
          slug
          name
          description
        }
      }
    }
  }
}
```

**Comment by DomDew** [comment]
Hey @amityweb,

thank you for bringing this up!

Which version are you on? If you are on 2.0.3 I‘d suggest you either rollback to 2.0.2 or try passing the locale as a filter for the content type (on 2.0.3). I might have introduced a bug with the latest update.

**Comment by amityweb** [comment]
I was on 2.0.3 and downgraded to 2.0.2 but still get the issue. My npm install output states fuzzy-search needs yup but I need to install manually, so did that, but still same issue. 

If you think it really should be working then could it be an issue my side? Is the code correct? I have published all the pages. 

Using the filters returns empty/no results. Replacing es below with en works OK. 

```
query {
  search(query: "pro") {
    pages(filters: {locale: {eq:"es"}}) {
      data {
        attributes
        {
          slug
          title
        }
      }
    }
    products {
      data {
        attributes {
          slug
          name
          description
        }
      }
    }
  }
}
```

**Comment by DomDew** [comment]
@amityweb 

everything looks good on your end… It‘s weird that you are getting results for en but not for es. Could you check if you are getting results for any of Strapis default endpoints with that filter applied? 

Unfortunately I can‘t troubleshoot right now and am sorry that I can‘t be of much help… I will get back to this as soon as possible. If the Strapi endpoints also return no results, you could try rolling back to an earlier v1 version. I had the locale filtering (by passing locale arg) running since the earliest versions…

**Comment by amityweb** [comment]
Going to the following on Strapi does return the other language pages yes:
`http://localhost:3000/api/pages?locale=es`

Calling pages using GraphQL also works, as the pages themselves work on the website, they show the other language content.

Its just via fuzzy-search api there's no results. 

Thanks

**Comment by DomDew** [comment]
Fixed in [v2.0.5](https://www.npmjs.com/package/strapi-plugin-fuzzy-search/v/2.0.5)

@all-contributors please add @amityweb for bug

**Comment by amityweb** [comment]
Fantastic, I think both issues are solved now, the 4 character locale fr-FR and search results also show. Thanks a lot.
